server/controllers/uploadImage.js

```diff
@@ -0,0 +1,52 @@
+// @route POST /uploadImage/upload
+// @desc upload single image
+// @access Public
+
+const asyncHandler = require("express-async-handler");
+const { dataUri } = require("../middleware/multer");
+const { cloudinary } = require("../utils/cloudinary");
+
+exports.uploadImage = asyncHandler(async (req, res, next) => {
+  try {
+    const file = dataUri(req.file).content;
+    const uploadResponse = await cloudinary.uploader.upload(file);
+    const url = uploadResponse.url;
```

Comment:
Just working on integration with FE now... I think we should update the photo attribute in the profile model once we get the URL.

Reply:
I thought about that, but wouldn't that be a separate route? Because there are more instances for image upload than just profile photo like messaging

Reply:
Oh - I get where you're coming from now. Maybe a separate route after all if we're using this for something else. I might be able to add that in integration. Thanks!